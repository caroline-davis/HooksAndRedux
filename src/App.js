import React, {useState, useEffect} from 'react';
import Input from './components/Input.js';
import Save from './components/Save.js';
import { getAddress } from './apis/api.js';
import './App.css';


function App() {

  const [name, setName] = useState("")
  const [names, setNames] = useState([])

  const [age, setAge] = useState("")
  const [ages, setAges] = useState([])

  const [food, setFood] = useState("")
  const [foods, setFoods] = useState([])

  const [address, setAddress] = useState([])
  const[loading, setLoading] = useState([])

  useEffect(() => {
    async function loadAddress () {
      const address = await getAddress()
      setAddress(address)
    }
    loadAddress()
   }, [])

  return (
    <div className="container">

      <Input placeholder="name"
             value={name}
             onChange={setName}>
      </Input>

      <Input placeholder="age"
             value={age}
             onChange={setAge}>
      </Input>

      <Input placeholder="food"
             value={food}
             onChange={setFood}> 
      </Input>
   
    <div>
     
      <Save handleClick={setNames}
            curVal={names}
            newVal={name}
            field="Name">
      </Save>
      <Save handleClick={setAges}
            curVal={ages}
            newVal={age}
            field="Age">
      </Save>

      <button onClick={()=> setFoods([...foods, food])}>Save Food</button>

    </div>

    {names.map((n, index) => 
      <div key={index}>
        Name: {n}
      </div>
    )}

    {ages.map((a, index) => 
    <div key={index}>
      Age: {a}
    </div>
    )}

    {foods.map((m, index) =>
     <div key={index}>
       Food: {m}
     </div>
    )}
    <div>
      {console.log(address.city)}
      {address.city} {address.country}
    </div>

    </div>

  );
}

export default App;

