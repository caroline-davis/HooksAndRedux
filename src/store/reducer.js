const initialState = {
    weather: ""
}

const reducer = (state = initialState, action) => {

    switch (action.type) {
        case 'HOT':
            return {
                ...state,
                weather: state.weather + "hot"
            }
        case 'COLD':
            return {
                ...state,
                weather: state.weather + "cold"
        }
        default:
            return state;
    }
};

export default reducer;



    
