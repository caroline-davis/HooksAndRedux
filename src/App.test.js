import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { createStore } from "redux";
import { Provider } from 'react-redux';
import reducer from './store/reducer';

const store = createStore(reducer);

describe('App component', () => {
  it('matches the snapshot', () => {
    const tree  = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
