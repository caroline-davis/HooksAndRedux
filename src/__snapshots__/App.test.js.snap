// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="container"
>
  <input
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="age"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="food"
    type="text"
    value=""
  />
  <div>
    <button
      className="saveBtn"
      onClick={[Function]}
    >
      Save 
      Name
    </button>
    <button
      className="saveBtn"
      onClick={[Function]}
    >
      Save 
      Age
    </button>
    <button
      onClick={[Function]}
    >
      Save Food
    </button>
  </div>
  <div>
    <div>
      Loading...
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      is it hot?
    </button>
    <button
      onClick={[Function]}
    >
      is it cold?
    </button>
    <div>
      
    </div>
  </div>
</div>
`;
